recipes/dev/generate_v2.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import itertools
+import sys
+import time
+from typing import Any, Dict, List
+
+import torch
+from omegaconf import DictConfig
+
+from torchtune import config, training, utils
+from torchtune.data import (
+    left_pad_sequence,
+    load_image,
+    Message,
+    padded_collate_tiled_images_and_mask,
+)
+
+from torchtune.generation import sample
+
+from torchtune.modules.transforms import Transform
+
+
+class SingleTurnYAMLToMessages(Transform):
+    """
+    Converts a single turn conversation in YAML format to a list of messages.
+    """
+
+    def __call__(self, prompt: Dict[str, Any]) -> List[Message]:
+        messages = []
+        for role, content in prompt.items():
+            if isinstance(content, str):
+                new_content = [{"type": "text", "content": content}]
+            else:
+                assert (
+                    "image" in content.keys()
+                ), "Multiple entries per role expect an image key"
+                image_loc = content["image"]
+                image = load_image(image_loc)
+                new_content = [
+                    {"type": "image", "content": image},
+                    {"type": "text", "content": content["text"]},
+                ]
+            messages.append(Message(role=role, content=new_content))
+        return messages
+
+
+def batch_to_device(batch: dict, device: torch.device) -> None:
```

Comment:
Put in training/ or something?

Reply:
Yeah @pbontrager is going to add this in his PR. This code will be deleted here.